src/main/java/org/truffleruby/extra/ffi/PointerNodes.java

```diff
@@ -156,23 +161,29 @@ public DynamicObject readString(DynamicObject pointer, int length) {
     public static abstract class PointerReadStringToNullPrimitiveNode extends PrimitiveArrayArgumentsNode {
 
         @Specialization(guards = "isNullPointer(pointer)")
-        public DynamicObject readNullPointer(DynamicObject pointer) {
-            return createString(RopeConstants.EMPTY_ASCII_8BIT_ROPE);
+        public DynamicObject readNullPointer(DynamicObject pointer,
+                                             @Cached("create()") AllocateObjectNode allocateObjectNode) {
+            return allocateObjectNode.allocate(coreLibrary().getStringClass(), Layouts.STRING.build(false, false, RopeConstants.EMPTY_ASCII_8BIT_ROPE));
```

Comment:
This is extremely verbose for allocating an empty binary String.
Any idea how to shorten it?
`false, false` seems particularly redundant as we allocate a new String here and in most cases they will be tainted=false, frozen=false.
Maybe `AllocateObjectNode.createString(Rope)` ?